gno.land/pkg/gnoland/types.go

```diff
@@ -19,7 +19,92 @@ func ProtoGnoAccount() std.Account {
 	return &GnoAccount{}
 }
 
+// GnoGenesis defines the gno genesis API,
+// adopted by differing genesis state implementations
+type GnoGenesis interface {
+	// GenesisBalances returns the genesis balances associated
+	// with the Gno genesis state
+	GenesisBalances() []Balance
+
+	// GenesisTxs returns the genesis transactions associated
+	// with the Gno genesis state
+	GenesisTxs() []GenesisTx
+}
+
+type GenesisTx interface {
+	// Tx returns the standard TM2 transaction
+	Tx() std.Tx
+
+	// Metadata returns the metadata tied
+	// to the tx, if any
+	Metadata() *GenesisTxMetadata
```

Comment:
nit: I would not return a pointer here, as it would simplify usage and considerations. Additionally, based on the MetadataTX implementation, allowing the caller to modify the underlying metadata could be problematic and may lead to unexpected side effects (especially in an in-memory context).